### PR TITLE
Functional tests - Add new test 'FO > Home page > Display some products'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'FO > Home page > Display some products'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | Nightly run => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/3904758156
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/pages/FO/home/index.ts [hunk 1]
```diff
@@ -286,12 +298,117 @@ class Home extends FOBasePage {
...
    * @returns {Promise<string>}
    */
-  getPopularProductTitle(page: Page): Promise<string> {
-    return this.getTextContent(page, this.popularProductTitle);
+  async getBlockTitle(page: Page, blockID: number = 1): Promise<string> {
+    let columnSelector: string;
+
+    switch (blockID) {
+      case 1:
+        columnSelector = this.productsBlockTitle(2) as string;
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.ts]: why we need casts here ?

File: tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import pricesDropPage from '@pages/FO/home/pricesDrop';
+import newProductsPage from '@pages/FO/home/newProducts';
+import bestSellersPage from '@pages/FO/home/bestSellers';
+
+import {expect} from 'chai';
+import type {BrowserContext, Page} from 'playwright';
+
+const baseContext: string = 'functional_FO_homePage_displaySomeProducts';
+
+describe('FO - Home Page : Display some products', async () => {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts:25]: Please some description of the test before first describe

Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts:25]: th done :+1:

File: tests/UI/pages/FO/home/index.ts [hunk 3]
```diff
@@ -286,12 +298,117 @@ class Home extends FOBasePage {
...
+   * Get products block number
+   * @param blockID {number} The block number in the page
+   * @param page {Page} Browser tab
+   */
+  async getProductsBlockNumber(page: Page, blockID: number = 1): Promise<number> {
+    let columnSelector: string;
+
+    switch (blockID) {
+      case 1:
+        columnSelector = this.productsBlockDiv(2) as string;
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.ts]: Same as here

File: tests/UI/pages/FO/home/index.ts [hunk 4]
```diff
@@ -286,12 +298,117 @@ class Home extends FOBasePage {
...
+   * @param page {Page} Browser tab
+   * @param blockID {number} The block number in the page
+   * @return {Promise<void>}
+   */
+  async goToAllProductsBlockPage(page: Page, blockID: number = 1): Promise<void> {
+    let columnSelector: string;
+
+    switch (blockID) {
+      case 1:
+        columnSelector = this.allProductsBlockLink(2) as string;
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.ts]: Same here

File: tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      await expect(isHomePage, 'Home page is not displayed').to.be.true;
+    });
+  });
+
+  describe('Check best sellers block', async () => {
+    it('should check best sellers block title', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkBestSellerProducts', baseContext);
+
+      const popularProductTitle = await homePage.getBlockTitle(page, 4);
+      await expect(popularProductTitle).to.equal('Best Sellers');
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts:173]: Can't use `BestSales.pageTitle` ?

File: tests/UI/pages/FO/home/bestSellers.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+// Import pages
+import FOBasePage from '@pages/FO/FObasePage';
```
Comment (Progi1984) [inline tests/UI/pages/FO/home/bestSellers.ts]: Isn't it a duplicate to `tests/UI/pages/FO/bestSales/index.ts` file ?

File: tests/UI/pages/FO/home/newProducts.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+// Import pages
+import FOBasePage from '@pages/FO/FObasePage';
```
Comment (Progi1984) [inline tests/UI/pages/FO/home/newProducts.ts]: Duplicate of `tests/UI/pages/FO/newProducts/index.ts` ?

File: tests/UI/pages/FO/home/pricesDrop.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+// Import pages
+import FOBasePage from '@pages/FO/FObasePage';
```
Comment (Progi1984) [inline tests/UI/pages/FO/home/pricesDrop.ts]: Duplicate of `tests/UI/pages/FO/pricesDrop/index.ts` ?

**Review by Progi1984 — CHANGES_REQUESTED** [review]
It seems you have some duplicate file with existing ;)

File: tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts (hunk 5 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/functional/FO/06_homePage/02_displaySomeProducts.ts:173]: It's not a page title, it's a block title in home page :)

File: tests/UI/pages/FO/home/bestSellers.ts (hunk 6 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/home/bestSellers.ts]: Yes, nice catch thank you :)

File: tests/UI/pages/FO/home/newProducts.ts (hunk 7 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/home/newProducts.ts]: Done :+1:

File: tests/UI/pages/FO/home/pricesDrop.ts (hunk 8 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/home/pricesDrop.ts]: Done :+1:

Comment (Progi1984) [reply tests/UI/pages/FO/home/pricesDrop.ts]: Could you remove this one @nesrineabdmouleh ?

File: tests/UI/pages/FO/home/index.ts [hunk 9]
```diff
@@ -43,12 +45,18 @@ class Home extends FOBasePage {
 
   public readonly allProductLink: string;
 
+  public readonly allProductsBlockLink: (blockId: number) => string;
```
Comment (Progi1984) [inline tests/UI/pages/FO/home/index.ts]:
```suggestion
  private readonly allProductsBlockLink: (blockId: number) => string;
```

**Review by Progi1984 — COMMENTED** [review]
Two last feedbacks ;)

File: tests/UI/pages/FO/home/index.ts (hunk 9 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/FO/home/index.ts]: done thx :)

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**